cmdtest.go

```diff
@@ -335,10 +336,11 @@ func (ts *TestSuite) update(t *testing.T) {
 	for _, tf := range ts.files {
 		t.Run(strings.TrimSuffix(tf.filename, ".ct"), func(t *testing.T) {
 			tmpfile, err := tf.updateToTemp()
+			defer tmpfile.Cleanup()
```

Comment:
I think this should be moved after the `if err != nil` check. `tmpfile` can be nil if err is not.

Also, check the error from Cleanup and fail it it's not nil.

Reply:
Oops, my bad. I deferred the cleanup before checking `err` because the file also needs to be cleaned up when there is a writing error, not only in successful scenarios.

Therefore, I fixed it and rebased in https://github.com/google/go-cmdtest/pull/4/commits/b1b394ee42be715f4f7eafd5238275efe116b453.

Now when there's a write error, `(*TestSuite).write` returns both the file pointer and the error, so we can defer a cleanup whenever `tmpfile` is not `nil`.